### ISSUE TITLE
Undefined symbols for architecture arm64:   "_FIRFirebaseVersion", referenced from:       +[FIRAnalytics topLevelVersion] in FirebaseAnalytics[10](FIRAnalytics.o)

### ISSUE THREAD

**Issue by Sagar1upa** [firebase/firebase-ios-sdk#13468]
### Description

After updating from 9 version to 11
Has error project didn't build

![image](https://github.com/user-attachments/assets/78310fab-2272-4fb3-80c9-37cc4bb718a0)

### Firebase SDK Version

11.0

### Xcode Version

15.3

### Installation Method

Swift Package Manager

### Firebase Product(s)

Analytics, DynamicLinks, In-App Messaging, Installations, Messaging

### Targeted Platforms

iOS

### Relevant Log Output

```shell
ld: warning: Could not find or use auto-linked framework 'CoreAudioTypes': framework 'CoreAudioTypes' not found
Undefined symbols for architecture arm64:
  "_FIRFirebaseVersion", referenced from:
      +[FIRAnalytics topLevelVersion] in FirebaseAnalytics[10](FIRAnalytics.o)
  "_OBJC_CLASS_$_FIRApp", referenced from:
       in FirebaseAnalytics[10](FIRAnalytics.o)
  "_OBJC_CLASS_$_FIRComponent", referenced from:
       in FirebaseAnalytics[11](FIRAnalyticsConnector.o)
  "_OBJC_CLASS_$_GULAppEnvironmentUtil", referenced from:
       in GoogleAppMeasurement[5](APMAEU.o)
  "_kFIRLoggerAnalytics", referenced from:
      +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics[10](FIRAnalytics.o)
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Replace this line with the contents of your Package.resolved.

```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by paulb777** [comment]
There is likely something stale in the Xcode project, Package Manager installation, or DerivedData.

We'll probably need a fully reproducible example to help.

**Comment by Jride** [comment]
@paulb777 I am updating from version 10.16.0 to 10.29.0 and I am seeing the same linking issues, but only when compiling for the unit test bundle. This happens on our CI as well, which is a fresh instance each time so can't be anything stale there.

```
Undefined symbols for architecture arm64:
  "_FIRFirebaseVersion", referenced from:
⚠️       +[FIRAnalytics topLevelVersion] in FirebaseAnalytics[arm64][10](FIRAnalytics.o)
❌   "_FIRInstallationIDDidChangeNotification", referenced from:
⚠️       +[FIRAnalytics observeFirebaseInstallationIDChanges] in FirebaseAnalytics[arm64][10](FIRAnalytics.o)
❌   "_GULIsLoggableLevel", referenced from:
⚠️       -[APMMonitor isLoggableLevel:] in GoogleAppMeasurement[arm64][68](APMMonitor.o)
❌   "_GULLogBasic", referenced from:
⚠️       -[APMASLLogger logMessage:logTag:messageCode:withLogLevel:] in GoogleAppMeasurement[arm64][12](APMASLLogger.o)
❌   "_GULLogError", referenced from:
⚠️       -[APMPersistentDictionary initWithFileName:] in GoogleAppMeasurement[arm64][261](APMPersistentDictionary.o)
⚠️       _APMWriteDictionaryToURL in GoogleAppMeasurement[arm64][261](APMPersistentDictionary.o)
⚠️       -[APMUserDefaults synchronize] in GoogleAppMeasurement[arm64][315](APMUserDefaults.o)
❌   "_GULLogInfo", referenced from:
⚠️       ___44+[UIViewController(APMScreenClassName) load]_block_invoke in GoogleAppMeasurement[arm64][334](UIViewController+APMScreenClassName.o)
❌   "_GULLogWarning", referenced from:
⚠️       +[APMMeasurement sharedInstance] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
⚠️       -[APMPersistentDictionary objectForKey:] in GoogleAppMeasurement[arm64][261](APMPersistentDictionary.o)
⚠️       -[APMPersistentDictionary setObject:forKey:] in GoogleAppMeasurement[arm64][261](APMPersistentDictionary.o)
⚠️       -[APMUserDefaults objectForKey:] in GoogleAppMeasurement[arm64][315](APMUserDefaults.o)
⚠️       -[APMUserDefaults setObject:forKey:] in GoogleAppMeasurement[arm64][315](APMUserDefaults.o)
⚠️       -[APMUserDefaults synchronize] in GoogleAppMeasurement[arm64][315](APMUserDefaults.o)
⚠️       ___44+[UIViewController(APMScreenClassName) load]_block_invoke in GoogleAppMeasurement[arm64][334](UIViewController+APMScreenClassName.o)
❌   "_GULSetLoggerLevel", referenced from:
⚠️       -[APMMonitor setDebugModeEnabled:] in GoogleAppMeasurement[arm64][68](APMMonitor.o)
⚠️       -[APMMonitor setDebugModeEnabled:] in GoogleAppMeasurement[arm64][68](APMMonitor.o)
⚠️       -[APMMonitor setVerboseLoggingEnabled:] in GoogleAppMeasurement[arm64][68](APMMonitor.o)
❌   "_OBJC_CLASS_$_FIRApp", referenced from:
⚠️        in FirebaseAnalytics[arm64][10](FIRAnalytics.o)
❌   "_OBJC_CLASS_$_FIRComponent", referenced from:
⚠️        in FirebaseAnalytics[arm64][11](FIRAnalyticsConnector.o)
❌   "_OBJC_CLASS_$_FIRInstallations", referenced from:
⚠️        in FirebaseAnalytics[arm64][10](FIRAnalytics.o)
❌   "_OBJC_CLASS_$_GULAppDelegateSwizzler", referenced from:
⚠️        in GoogleAppMeasurement[arm64][7](APMAnalytics.o)
❌   "_OBJC_CLASS_$_GULAppEnvironmentUtil", referenced from:
⚠️        in GoogleAppMeasurement[arm64][5](APMAEU.o)
❌   "_OBJC_CLASS_$_GULMutableDictionary", referenced from:
⚠️        in GoogleAppMeasurement[arm64][35](APMEExperiment.o)
❌   "_OBJC_CLASS_$_GULNetwork", referenced from:
⚠️        in GoogleAppMeasurement[arm64][7](APMAnalytics.o)
❌   "_OBJC_CLASS_$_GULSwizzler", referenced from:
⚠️        in GoogleAppMeasurement[arm64][334](UIViewController+APMScreenClassName.o)
❌   "_kFIRLoggerAnalytics", referenced from:
⚠️       +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics[arm64][10](FIRAnalytics.o)
❌   "_kGULNetworkHTTPStatusCodeCannotAcceptTraffic", referenced from:
⚠️       -[APMETaskManager handleFetchingExperimentsResponse:data:error:] in GoogleAppMeasurement[arm64][42](APMETaskManager.o)
⚠️       -[APMMeasurement networkUploadCompletionHandlerWithResponse:error:] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
⚠️       -[APMMeasurement networkRemoteConfigFetchCompletionHandler:data:error:] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
❌   "_kGULNetworkHTTPStatusCodeNotFound", referenced from:
⚠️       -[APMMeasurement networkRemoteConfigFetchCompletionHandler:data:error:] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
❌   "_kGULNetworkHTTPStatusCodeNotModified", referenced from:
⚠️       ___37-[APMMeasurement retrievePSMResponse]_block_invoke in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
⚠️       -[APMMeasurement networkRemoteConfigFetchCompletionHandler:data:error:] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
⚠️       -[APMMeasurement(SKAN) handleSKANFetchResponse:data:error:] in GoogleAppMeasurement[arm64][66](APMMeasurement+SKAN.o)
⚠️       ___46-[APMMeasurement(URL) requestDeferredDeepLink]_block_invoke in GoogleAppMeasurement[arm64][67](APMMeasurement+URL.o)
❌   "_kGULNetworkHTTPStatusCodeUnavailable", referenced from:
⚠️       -[APMETaskManager handleFetchingExperimentsResponse:data:error:] in GoogleAppMeasurement[arm64][42](APMETaskManager.o)
⚠️       -[APMMeasurement networkUploadCompletionHandlerWithResponse:error:] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
⚠️       -[APMMeasurement networkRemoteConfigFetchCompletionHandler:data:error:] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
❌   "_kGULNetworkHTTPStatusNoContent", referenced from:
⚠️       -[APMETaskManager handleFetchingExperimentsResponse:data:error:] in GoogleAppMeasurement[arm64][42](APMETaskManager.o)
⚠️       -[APMMeasurement networkUploadCompletionHandlerWithResponse:error:] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
⚠️       ___37-[APMMeasurement retrievePSMResponse]_block_invoke in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
⚠️       -[APMMeasurement networkRemoteConfigFetchCompletionHandler:data:error:] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
⚠️       ___46-[APMMeasurement(URL) requestDeferredDeepLink]_block_invoke in GoogleAppMeasurement[arm64][67](APMMeasurement+URL.o)
❌   "_kGULNetworkHTTPStatusOK", referenced from:
⚠️       -[APMETaskManager handleFetchingExperimentsResponse:data:error:] in GoogleAppMeasurement[arm64][42](APMETaskManager.o)
⚠️       -[APMMeasurement networkUploadCompletionHandlerWithResponse:error:] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
⚠️       ___37-[APMMeasurement retrievePSMResponse]_block_invoke in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
⚠️       -[APMMeasurement networkRemoteConfigFetchCompletionHandler:data:error:] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
⚠️       -[APMMeasurement(SKAN) handleSKANFetchResponse:data:error:] in GoogleAppMeasurement[arm64][66](APMMeasurement+SKAN.o)
⚠️       ___46-[APMMeasurement(URL) requestDeferredDeepLink]_block_invoke in GoogleAppMeasurement[arm64][67](APMMeasurement+URL.o)
❌   "_pb_decode", referenced from:
⚠️       _apmpb_decode_ad_campaign_info in GoogleAppMeasurement[arm64][78](apmpb_ad_campaign_info_decoder.o)
⚠️       _apmpb_decode_app_property in GoogleAppMeasurement[arm64][81](apmpb_app_property_decoder.o)
⚠️       _apmpb_decode_audience in GoogleAppMeasurement[arm64][84](apmpb_audience_decoder.o)
⚠️       _apmpb_decode_audience in GoogleAppMeasurement[arm64][84](apmpb_audience_decoder.o)
⚠️       _apmpb_decode_audience_leaf_filter_result in GoogleAppMeasurement[arm64][87](apmpb_audience_leaf_filter_result_decoder.o)
⚠️       _apmpb_decode_audience_leaf_filter_result in GoogleAppMeasurement[arm64][87](apmpb_audience_leaf_filter_result_decoder.o)
⚠️       _apmpb_decode_client_property in GoogleAppMeasurement[arm64][90](apmpb_client_property_decoder.o)
⚠️       _apmpb_decode_client_property in GoogleAppMeasurement[arm64][90](apmpb_client_property_decoder.o)
❌   "_pb_decode_varint", referenced from:
⚠️       _nano_decode_repeated_uint64 in GoogleAppMeasurement[arm64][319](decoder_common.o)
❌   "_pb_encode", referenced from:
⚠️       _nano_two_pass_encoding in GoogleAppMeasurement[arm64][320](encoder_common.o)
⚠️       _nano_two_pass_encoding in GoogleAppMeasurement[arm64][320](encoder_common.o)
❌   "_pb_encode_string", referenced from:
⚠️       _nano_encode_string in GoogleAppMeasurement[arm64][320](encoder_common.o)
❌   "_pb_encode_submessage", referenced from:
⚠️       _apmpb_encode_repeated_ad_campaign_info in GoogleAppMeasurement[arm64][79](apmpb_ad_campaign_info_encoder.o)
⚠️       _apmpb_encode_repeated_app_property in GoogleAppMeasurement[arm64][82](apmpb_app_property_encoder.o)
⚠️       _apmpb_encode_repeated_audience in GoogleAppMeasurement[arm64][85](apmpb_audience_encoder.o)
⚠️       _apmpb_encode_repeated_audience_leaf_filter_result in GoogleAppMeasurement[arm64][88](apmpb_audience_leaf_filter_result_encoder.o)
⚠️       _apmpb_encode_repeated_client_property in GoogleAppMeasurement[arm64][91](apmpb_client_property_encoder.o)
⚠️       _apmpb_encode_repeated_consent_config in GoogleAppMeasurement[arm64][94](apmpb_consent_config_encoder.o)
⚠️       _apmpb_encode_repeated_consent_default_entry in GoogleAppMeasurement[arm64][97](apmpb_consent_default_entry_encoder.o)
❌   "_pb_encode_tag_for_field", referenced from:
⚠️       _apmpb_encode_repeated_ad_campaign_info in GoogleAppMeasurement[arm64][79](apmpb_ad_campaign_info_encoder.o)
⚠️       _apmpb_encode_repeated_app_property in GoogleAppMeasurement[arm64][82](apmpb_app_property_encoder.o)
⚠️       _apmpb_encode_repeated_audience in GoogleAppMeasurement[arm64][85](apmpb_audience_encoder.o)
⚠️       _apmpb_encode_repeated_audience_leaf_filter_result in GoogleAppMeasurement[arm64][88](apmpb_audience_leaf_filter_result_encoder.o)
⚠️       _apmpb_encode_repeated_client_property in GoogleAppMeasurement[arm64][91](apmpb_client_property_encoder.o)
⚠️       _apmpb_encode_repeated_consent_config in GoogleAppMeasurement[arm64][94](apmpb_consent_config_encoder.o)
⚠️       _apmpb_encode_repeated_consent_default_entry in GoogleAppMeasurement[arm64][97](apmpb_consent_default_entry_encoder.o)
❌   "_pb_encode_varint", referenced from:
⚠️       _nano_encode_repeated_uint64 in GoogleAppMeasurement[arm64][320](encoder_common.o)
❌   "_pb_istream_from_buffer", referenced from:
⚠️       _apmpb_decode_ad_campaign_info in GoogleAppMeasurement[arm64][78](apmpb_ad_campaign_info_decoder.o)
⚠️       _apmpb_decode_app_property in GoogleAppMeasurement[arm64][81](apmpb_app_property_decoder.o)
⚠️       _apmpb_decode_audience in GoogleAppMeasurement[arm64][84](apmpb_audience_decoder.o)
⚠️       _apmpb_decode_audience in GoogleAppMeasurement[arm64][84](apmpb_audience_decoder.o)
⚠️       _apmpb_decode_audience_leaf_filter_result in GoogleAppMeasurement[arm64][87](apmpb_audience_leaf_filter_result_decoder.o)
⚠️       _apmpb_decode_audience_leaf_filter_result in GoogleAppMeasurement[arm64][87](apmpb_audience_leaf_filter_result_decoder.o)
⚠️       _apmpb_decode_client_property in GoogleAppMeasurement[arm64][90](apmpb_client_property_decoder.o)
⚠️       _apmpb_decode_client_property in GoogleAppMeasurement[arm64][90](apmpb_client_property_decoder.o)
❌   "_pb_ostream_from_buffer", referenced from:
⚠️       _nano_two_pass_encoding in GoogleAppMeasurement[arm64][320](encoder_common.o)
❌   "_pb_read", referenced from:
⚠️       _apmpb_decode_repeated_ad_campaign_info in GoogleAppMeasurement[arm64][78](apmpb_ad_campaign_info_decoder.o)
⚠️       _apmpb_decode_repeated_ad_campaign_info in GoogleAppMeasurement[arm64][78](apmpb_ad_campaign_info_decoder.o)
⚠️       _apmpb_decode_repeated_app_property in GoogleAppMeasurement[arm64][81](apmpb_app_property_decoder.o)
⚠️       _apmpb_decode_repeated_app_property in GoogleAppMeasurement[arm64][81](apmpb_app_property_decoder.o)
⚠️       _apmpb_decode_repeated_audience in GoogleAppMeasurement[arm64][84](apmpb_audience_decoder.o)
⚠️       _apmpb_decode_repeated_audience in GoogleAppMeasurement[arm64][84](apmpb_audience_decoder.o)
⚠️       _apmpb_decode_repeated_audience_leaf_filter_result in GoogleAppMeasurement[arm64][87](apmpb_audience_leaf_filter_result_decoder.o)
⚠️       _apmpb_decode_repeated_audience_leaf_filter_result in GoogleAppMeasurement[arm64][87](apmpb_audience_leaf_filter_result_decoder.o)
❌ ld: symbol(s) not found for architecture arm64
❌ clang: error: linker command failed with exit code 1 (use -v to see invocation)
⚠️ Skipping duplicate build file in Compile Sources build phase: /Users/vagrant/git/iOS/ITV Hub/Services/ABTesting/API/Amplitude/Amplitude+ExperimentSDK+ABTestIdentifier.swift (in target 'ITVHub_iOS' from project 'ITVHub_iOS')
Testing failed:
	Undefined symbol: _FIRFirebaseVersion
	Undefined symbol: _FIRInstallationIDDidChangeNotification
	Undefined symbol: _GULIsLoggableLevel
	Undefined symbol: _GULLogBasic
	Undefined symbol: _GULLogError
	Undefined symbol: _GULLogInfo
	Undefined symbol: _GULLogWarning
	Undefined symbol: _GULSetLoggerLevel
	Undefined symbol: _OBJC_CLASS_$_FIRApp
	Undefined symbol: _OBJC_CLASS_$_FIRComponent
	Undefined symbol: _OBJC_CLASS_$_FIRInstallations
	Undefined symbol: _OBJC_CLASS_$_GULAppDelegateSwizzler
	Undefined symbol: _OBJC_CLASS_$_GULAppEnvironmentUtil
	Undefined symbol: _OBJC_CLASS_$_GULMutableDictionary
	Undefined symbol: _OBJC_CLASS_$_GULNetwork
	Undefined symbol: _OBJC_CLASS_$_GULSwizzler
	Undefined symbol: _kFIRLoggerAnalytics
	Undefined symbol: _kGULNetworkHTTPStatusCodeCannotAcceptTraffic
	Undefined symbol: _kGULNetworkHTTPStatusCodeNotFound
	Undefined symbol: _kGULNetworkHTTPStatusCodeNotModified
	Undefined symbol: _kGULNetworkHTTPStatusCodeUnavailable
	Undefined symbol: _kGULNetworkHTTPStatusNoContent
	Undefined symbol: _kGULNetworkHTTPStatusOK
	Undefined symbol: _pb_decode
	Undefined symbol: _pb_decode_varint
	Undefined symbol: _pb_encode
	Undefined symbol: _pb_encode_string
	Undefined symbol: _pb_encode_submessage
	Undefined symbol: _pb_encode_tag_for_field
	Undefined symbol: _pb_encode_varint
	Undefined symbol: _pb_istream_from_buffer
	Undefined symbol: _pb_ostream_from_buffer
	Undefined symbol: _pb_read
	Linker command failed with exit code 1 (use -v to see invocation)
	Testing cancelled because the build failed.
```

**Comment by Nonouf** [comment]
We’re having exactly the same issue. I don’t know if it’s related but it could be because we’re building using Rosetta.
```
Undefined symbols for architecture x86_64:
  "_APMUserDataFieldHashedEmailAddress", referenced from:
      +[FIRAnalytics initiateOnDeviceConversionMeasurementWithHashedEmailAddress:] in FirebaseAnalytics[x86_64][10](FIRAnalytics.o)
  "_APMUserDataFieldHashedPhoneNumber", referenced from:
      +[FIRAnalytics initiateOnDeviceConversionMeasurementWithHashedPhoneNumber:] in FirebaseAnalytics[x86_64][10](FIRAnalytics.o)
  "_GULLogBasic", referenced from:
      -[APMASLLogger logMessage:logTag:messageCode:withLogLevel:] in GoogleAppMeasurement[x86_64][11](APMASLLogger.o)
  "_GULLogError", referenced from:
      -[APMPersistentDictionary initWithFileName:] in GoogleAppMeasurement[x86_64][231](APMPersistentDictionary.o)
      _APMWriteDictionaryToURL in GoogleAppMeasurement[x86_64][231](APMPersistentDictionary.o)
      -[APMUserDefaults synchronize] in GoogleAppMeasurement[x86_64][282](APMUserDefaults.o)
  "_GULLogInfo", referenced from:
      ___44+[UIViewController(APMScreenClassName) load]_block_invoke in GoogleAppMeasurement[x86_64][300](UIViewController+APMScreenClassName.o)
  "_GULLogWarning", referenced from:
      +[APMMeasurement sharedInstance] in GoogleAppMeasurement[x86_64][61](APMMeasurement.o)
      -[APMPersistentDictionary objectForKey:] in GoogleAppMeasurement[x86_64][231](APMPersistentDictionary.o)
      -[APMPersistentDictionary setObject:forKey:] in GoogleAppMeasurement[x86_64][231](APMPersistentDictionary.o)
      -[APMPersistentDictionary setObject:forKey:] in GoogleAppMeasurement[x86_64][231](APMPersistentDictionary.o)
      -[APMUserDefaults objectForKey:] in GoogleAppMeasurement[x86_64][282](APMUserDefaults.o)
      -[APMUserDefaults setObject:forKey:] in GoogleAppMeasurement[x86_64][282](APMUserDefaults.o)
      -[APMUserDefaults setObject:forKey:] in GoogleAppMeasurement[x86_64][282](APMUserDefaults.o)
      ...
ld: symbol(s) not found for architecture x86_64
```

**Comment by Jride** [comment]
Well we are not building for Rosetta so I think that might not be the issue. I am kind of stuck at the moment as we are updating the SDK to try and fix a crash we are experiencing at startup which I believe is linked to [this](https://github.com/firebase/firebase-ios-sdk/issues/12617) issue. It says that its been resolved in version 10.22.0 but downgrading to that still results in the above linking issues.

**Comment by paulb777** [comment]
We haven't been able to reproduce.  A fully reproducible example would be helpful.

**Comment by Jride** [comment]
Thanks for looking into this @paulb777 🙏 

Its hard to provide a reproducible example as spinning up a new project and linking Firebase seems to work just fine. Without providing access to the project (which sits in a private repo) i'm not sure what else I can provide. Happy to share any build settings we have in the project if that would help.

**Comment by paulb777** [comment]
One approach to make a reproducible example would be to copy the private project and try to delete as much as possible while the error still occurs.

**Comment by Jride** [comment]
@paulb777 Thanks for the suggestion. We managed to find a fix by updating our "Other Linker Flags" for our Unit Test target. Not sure which one of these flags were the culprit, but I just ended up doing a tidy up and match the setting that we have for our tvOS project which was building/linking the Firebase SDK just fine.

![coscreen-overlay](https://github.com/user-attachments/assets/d393bff8-ac5e-4a04-b870-5b4d623e9253)
